---
title: 'Data Fanny: comparison with expression and proteomic datasets'
author: "Marie Locard-Paulet"
date: "20 novembre 2015"
output: html_document
---

```{r pakages, echo=F}
require(ggplot2)
```

The comparison of Treg and Tconv has been done previously at the proteomic and transcriptomic level. Here I want to compare our dataset with previous works. I will work with the supplementary data of two papers:

* Foxp3 Transcription-Factor-Dependent and -Independent Regulation of the Regulatory T Cell Transcriptional Signature
Jonathan A. Hill,1,2 Markus Feuerer,1,2 Kaley Tash,1,2 Sokol Haxhinasto,1,2 Jasmine Perez,1 Rachel Melamed,1 Diane Mathis,1 and Christophe Benoist1,*
DOI 10.1016/j.immuni.2007.09.010

* Transcription Factor 7 Limits Regulatory T Cell Generation in the Thymus
Melanie M. Barra, David M. Richards, Jenny Hansson, Ann-Cathrin Hofer, Michael Delacher, Jan Hettinger, Jeroen Krijgsveld and Markus Feuerer
J Immunol published online 31 August 2015

The table with our MS results is the table "TotalAnalysisTconvTreg-A5.csv".

# Comparison with the expression dataset

Hill et al. 2007. The table used for the analysis is a supplementary table of the paper: "Table S1- Microarray probes that define the common Treg signature.  Individual probes were assigned a cluster membership, and correlation to Foxp3 as indicated in Fig. 6.  Mean expression values and fold-changes are also shown."

The input table of the expressin dataset is named "Signature FoxP3.csv" in the folder "ExpressionData". This table only contains information regarding the genes regulated in the dataset, and considered as markers of the Treg.

1. In the expression table, there can be several probes for a gene, I calculate the mean value per gene using the values in the column "TregSig".
2. I use a file for mapping the "Gene.Symbol" field in the expression table to the uniprot accessions in our table: "uniprot-yourlist%3AM201511262JG3RGOZU1_inputExpressionData".
3. In the expression table, I indicate wich genes are not present in our MS dataset by adding a column called "PresenceInOurDataset": "Present in our MS dataset" means we quantify the corresponding protein; "not in our MS dataset" means that we don't identify the corresponding protein in our MS dataset.
3. I then add a column named "MeanExpression" to our restult table that contains the mean expression observed for the corresponding protein in the expression dataset.

I plot the linear regression of upregulated genes and downregulated genes separately.

```{r Expression, echo = F}
tab <- read.csv("AllExperiments/allwoM4-NormAll/output-Score15/TotalAnalysisTconvTreg-A5.csv", stringsAsFactors = F)
exp <- read.delim("ExpressionData/signature Foxp3.csv", sep = ";", header = T)
tabmap <- sapply(tab$Majority.protein.IDs, function(x) strsplit(x, ";")[[1]][1])
tab <- cbind(tab, "mapping" = tabmap)

values <- sapply(unique(exp$Gene.Symbol), function(x) exp[exp$Gene.Symbol==x, "TregSig"])
names(values) <- unique(exp$Gene.Symbol)
means <- lapply(values, mean)
lmeans <- sapply(means, log2)

maptab <- read.delim("ExpressionData/uniprot-yourlist%3AM201511262JG3RGOZU1_inputExpressionData.tab", sep = "\t")
maptab2 <- maptab[maptab[,2] %in% tabmap,]

nam2 <- maptab2[,2][match(toupper(names(lmeans)), maptab2[,1])]


#tab1 <- merge(tab, exp, by.x = "Gene.names", by.y = "Gene.Symbol")
tab1 <- cbind(tab, "MeanExpression"= lmeans[match(tab$mapping, nam2)])
tab10 <- tab1
tab1 <- tab1[!is.na(tab1$MeanExpression),]
ngenes <- length(unique(tab1$Gene.names))

#colours for plot:
col <- rep("Not", nrow(tab1))
col[abs(tab1$MeanExpression) >= 2 | abs(tab1$FC) >= 2] <- "Above 2"
text <- rep(NA, nrow(tab1))
text[abs(tab1$MeanExpression) >= 2 | abs(tab1$FC) >= 2] <- tab1$Gene.names[abs(tab1$MeanExpression) >= 2 | abs(tab1$FC) >= 2]
tab1 <- cbind(tab1, "col" = col, "lab" = text)

#Separate the up and down:
tup <- tab1[tab1$MeanExpression > 0,]
td <- tab1[tab1$MeanExpression < 0,]
nup <- nrow(tup)
nd <- nrow(td)

lrup <- lm(tup$FC~tup$MeanExpression)
summary(lrup)
par(mfrow = c(2,2))
plot(lrup)

lrd <- lm(td$FC~td$MeanExpression)
summary(lrd)
par(mfrow = c(2,2))
plot(lrd)

tabg <- tup

gup <- ggplot(data = tabg, aes(y = FC, x = MeanExpression, label = lab)) + geom_vline(xintercept = 0) + geom_hline(yintercept = 0) + geom_smooth(method = lm, fullrange = TRUE) + geom_point(aes(col = col), alpha = 0.7, size = 3) + ggtitle("Expression of Treg markers Vs\nprotein Treg/Tconv ratio in our MS dataset") + theme_classic() + geom_vline(xintercept = log2(1.5), linetype = "dashed") + xlab("Mean expression: log2(Treg/Tconv)") + ylab("Proteomic data: log2(Treg/Tconv)") + scale_colour_manual(values = c("red", "Black")) + geom_text(col = "red", hjust = 0.5, vjust = -0.8, size = 4)
print(gup)

tabg <- td

gd <- ggplot(data = tabg, aes(y = FC, x = MeanExpression, label = lab)) + geom_vline(xintercept = 0) + geom_hline(yintercept = 0) + geom_smooth(method = lm, fullrange = TRUE) + geom_point(aes(col = col), alpha = 0.7, size = 3) + ggtitle("Expression of Treg markers Vs\nprotein Treg/Tconv ratio in our MS dataset") + theme_classic() + geom_vline(xintercept = log2(0.7), linetype = "dashed") + xlab("Mean expression: log2(Treg/Tconv)") + ylab("Proteomic data: log2(Treg/Tconv)") + scale_colour_manual(values = c("red", "Black")) + geom_text(col = "red", hjust = 0.5, vjust = -0.8, size = 4)
print(gd)

pdf("ExpressionData//FigureUp.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
gup
dev.off()

pdf("ExpressionData//FigureDown.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
gd
dev.off()

#write.csv(tab10, "TableAll_20151126.csv", row.names = F)

# Proteomic data regulated and not present in the expression dataset:
reg <- tab10[abs(tab10$FC) >= 1 & tab10$adjpval<=0.05,]
nprotnoexp <- nrow(reg[is.na(reg$MeanExpression),])


tabg <- reg[is.na(reg$MeanExpression),]
tabg <- tabg[order(tabg$FC),]
tabg$Gene.names <- factor(as.character(tabg$Gene.names), levels = as.character(tabg$Gene.names))

g <- ggplot(tabg, aes(y = FC, x = Gene.names)) + geom_hline(yintercept = 0) + geom_bar(stat = "identity", width = 0.7) + theme_bw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 5)) + ggtitle("Genes regulated in our MS dataset\nwith no expression data")
print(g)

pdf("ExpressionData//FigureNoExpData.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
g
dev.off()

# Make a column in the expression table that indicates if the gene is present in our MS dataset or not.
Pres <- rep("not in our MS dataset", nrow(exp))
Pres[toupper(exp$Gene.Symbol) %in% as.character(maptab2[,1])] <- "Present in our MS dataset"
exp <- cbind(exp, "PresenceInOurDataset" = Pres)
# write.csv(exp, "ExpressionData//signatureFoxp3-V2.csv", row.names = F)

table(Pres)
```

There are `r ngenes` unique genes that are present in the expression table and quantified in our MS dataset. There are `r nprotnoexp` proteins that are regulated in our dataset and not present in the expression dataset. These are presented in the previous figure.

The table "ExpressionData//signatureFoxp3-V2.csv" contains the initial expression data with an additional column stating if the gene is quantified in our MS dataset.

The table "TableAll_20151126" contains our result table with the additional columns, including a "mapping" column that is the first Majority protein ID that I use to merge the other datasets to ours.

# Comparison with the proteomic dataset

Supplemental Table 1: Calculated dataset of proteins differentially expressed between Treg cells and Tconv cells.

I use all the protein accessions in the column "Accession" as input in Uniprot ID mapper to get all possible protein accessions of these proteins: "uniprot-yourlist%3AM201511302JVB080VYT-ProtMSinput.tab".

```{r Proteomic, echo=F}
exp <- read.delim("Papier JI 2015/Sup Table 1.csv", sep = ";", header = T)

ProtMap <- sapply(as.character(exp$Accession), function(x) strsplit(x, ",", fixed = T)[[1]][1])
# write.csv(ProtMap, "Papier JI 2015//MappingNames.csv", row.names = F) # Protein list used for the ID mapping
# Issues with retrieval of the old Uniprot accessions so I use the gene names instead:
ProtMap <- as.character(exp$Gene.name)
# write.csv(ProtMap, "Papier JI 2015//MappingGeneNames.csv", row.names = F) # Protein list used for the ID mapping


maptab <- read.delim("Papier JI 2015/uniprot-yourlist%3AM2015113099G02LBX6Z-GenesMSinput.tab", sep = "\t")
maptab2 <- maptab[maptab[,2] %in% tab10$mapping,]

val <- exp$Average.log2.Ratio.Treg.Tconv[match(maptab2[,1], toupper(exp$Gene.name))]
val2 <- val[match(tab10$mapping, maptab2[,2])]


#tab1 <- merge(tab, exp, by.x = "Gene.names", by.y = "Gene.Symbol")
tab1 <- cbind(tab10, "ProteomicRatioTregTconv"= val2)

tab11 <- tab1
tab1 <- tab1[!is.na(tab1$ProteomicRatioTregTconv),]
ngenes <- length(unique(tab1$Gene.names))
nprot <- length(unique(tab1$mapping))

#colours for plot:
col <- rep("Not", nrow(tab1))
col[abs(tab1$ProteomicRatioTregTconv) >= 2 | abs(tab1$FC) >= 2] <- "Above 2"
text <- rep(NA, nrow(tab1))
text[abs(tab1$ProteomicRatioTregTconv) >= 2 | abs(tab1$FC) >= 2] <- tab1$Gene.names[abs(tab1$ProteomicRatioTregTconv) >= 2 | abs(tab1$FC) >= 2]
tab1 <- cbind(tab1, "col" = col, "lab" = text)

lr <- lm(tab1$FC~tab1$ProteomicRatioTregTconv)
summary(lr)
par(mfrow = c(2,2))
plot(lr)

tabg <- tab1

g <- ggplot(data = tabg, aes(y = FC, x = ProteomicRatioTregTconv, label = lab)) + geom_vline(xintercept = 0) + geom_hline(yintercept = 0) + geom_smooth(method = lm, fullrange = TRUE) + geom_point(aes(col = col), alpha = 0.7, size = 3) + ggtitle("Treg/Tconv ratio in the two MS datasets") + theme_classic() + xlab("Proteomic data from Barra el al. 2015: log2(Treg/Tconv)") + ylab("Proteomic data: log2(Treg/Tconv)") + scale_colour_manual(values = c("red", "Black")) + geom_text(col = "red", hjust = 0.5, vjust = -0.8, size = 4)
print(g)

pdf("Papier JI 2015//ComparisonProteomics.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
g
dev.off()

#write.csv(tab11, "TableAll_20151126.csv", row.names = F)

# Proteomic data regulated and not present in the Barra et al. dataset:
reg <- tab11[abs(tab10$FC) >= 1 & tab11$adjpval<=0.05,]
nprotnoexp <- nrow(reg[is.na(reg$ProteomicRatioTregTconv),])
nprotnoexp0 <- nrow(tab11[is.na(tab11$ProteomicRatioTregTconv),])


tabg <- reg[is.na(reg$ProteomicRatioTregTconv),]
tabg <- tabg[order(tabg$FC),]
tabg$Gene.names <- factor(as.character(tabg$Gene.names), levels = as.character(tabg$Gene.names))

g <- ggplot(tabg, aes(y = FC, x = Gene.names)) + geom_hline(yintercept = 0) + geom_bar(stat = "identity", width = 0.7) + theme_bw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 5)) + ggtitle("Proteins regulated in our MS dataset\nwith no quantification in Barra et al.")
print(g)

pdf("Papier JI 2015//FigureNoMSData.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
g
dev.off()

# Make a column in the expression table that indicates if the gene is present in our MS dataset or not.
Pres <- rep("not in our MS dataset", nrow(exp))
Pres[toupper(exp$Gene.name) %in% as.character(maptab2[,1])] <- "Present in our MS dataset"
exp <- cbind(exp, "PresenceInOurDataset" = Pres)
#write.csv(exp, "Papier JI 2015//Sup Table 1-V2.csv", row.names = F)
table(Pres)


nInThe3 <- nrow(tab11[!is.na(tab11$MeanExpression) & !is.na(tab11$ProteomicRatioTregTconv),])

tabg <- tab11[!is.na(tab11$MeanExpression) & !is.na(tab11$ProteomicRatioTregTconv),]
library(scatterplot3d)
par(mfrow = c(1,1))
scatterplot3d(x = tab11$FC, y = tab11$MeanExpression, z = tab11$ProteomicRatioTregTconv, type="h", main = "The 3 datasets")
library(rgl)
library(car)
scatter3d(x = tabg$FC, y = tabg$MeanExpression, z = tabg$ProteomicRatioTregTconv, type="h", main = "The 3 datasets", point.col = "blue", surface=FALSE, labels = tabg$Gene.names, id.n=nrow(tabg), sphere.size = 0.6)




col <- rep("0", nrow(tab1))
text <- rep(NA, nrow(tab1))
col[tab1$Gene.names%in% c("Eno2", "Foxp3", "Il2rb", "Nrp1", "Gbp4", "Klrg1")] <- "1"
text[tab1$Gene.names%in% c("Eno2", "Foxp3", "Il2rb", "Nrp1", "Gbp4", "Klrg1", "Pde3b", "Satb1", "Tcf7", "Pdlim4", "Ckb", "Padi2", "Dnm2", "Themis")] <-  c("Eno2", "Foxp3", "Il2rb", "Nrp1", "Gbp4", "Klrg1", "Pde3b", "Satb1", "Tcf7", "Pdlim4", "Ckb", "Padi2", "Dnm2", "Themis")[match(tab1$Gene.names[tab1$Gene.names %in% c("Eno2", "Foxp3", "Il2rb", "Nrp1", "Gbp4", "Klrg1", "Pde3b", "Satb1", "Tcf7", "Pdlim4", "Ckb", "Padi2", "Dnm2", "Themis")], c("Eno2", "Foxp3", "Il2rb", "Nrp1", "Gbp4", "Klrg1", "Pde3b", "Satb1", "Tcf7", "Pdlim4", "Ckb", "Padi2", "Dnm2", "Themis"))]
col[tab1$Gene.names%in% c("Pde3b", "Satb1", "Tcf7", "Pdlim4", "Ckb", "Padi2", "Dnm2")] <- "2"
col[tab1$Gene.names%in% c("Themis")] <- "3"
tabg <- data.frame(tab1, "colours"=col, "text" = text)

ggplot(data = tabg, aes(y = FC, x = ProteomicRatioTregTconv, col = colours, label = text)) + geom_vline(xintercept = 0) + geom_hline(yintercept = 0) + geom_point(alpha = 0.8, size = 3) + geom_smooth(data = tabg[tabg$colours=="0",], method = lm) + ggtitle("Linear regression of the two proteomic datasets") + scale_colour_manual(values = c("Black", "red", "blue", "forestgreen")) + geom_text(hjust = 1, vjust = 1) 

# Map markers:
map <- read.delim("AllExperiments/MappingMarkers.tab", header = T, row.names = NULL)
candidate <- as.character(unique(map$Entry[map$value=="candidate"]))
down <- as.character(unique(map$Entry[map$value=="down"]))
up <- as.character(unique(map$Entry[map$value=="up"]))

col <- rep("0", nrow(tab1))
text <- rep(NA, nrow(tab1))
col[tab1$Majority.protein.IDs%in% up] <- "1"
text[tab1$Majority.protein.IDs%in% c(up, down, candidate)] <-  tab1$Gene.names[tab1$Majority.protein.IDs %in% c(up, down, candidate)]
col[tab1$Majority.protein.IDs%in% down] <- "2"
col[tab1$Gene.names%in% c("Themis")] <- "3"
tabg <- data.frame(tab1, "colours"=col, "text" = text)

ggplot(data = tabg, aes(y = FC, x = ProteomicRatioTregTconv, col = colours, label = text)) + geom_vline(xintercept = 0) + geom_hline(yintercept = 0) + geom_point(alpha = 0.8, size = 3) + geom_smooth(data = tabg[tabg$colours=="0",], method = lm) + ggtitle("Linear regression of the two proteomic datasets") + scale_colour_manual(values = c("Black", "red", "blue", "forestgreen")) + geom_text(hjust = 1, vjust = 1) 


# Map regulated proteins:
candidate <- as.character(unique(map$Entry[map$value=="candidate"]))
down <- as.character(unique(tab1$Majority.protein.IDs[tab1$FC<=-1 & tab1$adjpval<=0.05]))
up <- as.character(unique(tab1$Majority.protein.IDs[tab1$FC>=1 & tab1$adjpval<=0.05]))

col <- rep("0", nrow(tab1))
text <- rep(NA, nrow(tab1))
col[tab1$Majority.protein.IDs%in% up] <- "1"
text[tab1$Majority.protein.IDs%in% c(up, down, candidate)] <-  tab1$Gene.names[tab1$Majority.protein.IDs %in% c(up, down, candidate)]
col[tab1$Majority.protein.IDs%in% down] <- "2"
col[tab1$Gene.names%in% c("Themis")] <- "3"
tabg <- data.frame(tab1, "colours"=col, "text" = text)

ggplot(data = tabg, aes(y = FC, x = ProteomicRatioTregTconv, col = colours, label = text)) + geom_vline(xintercept = 0) + geom_hline(yintercept = 0) + geom_point(alpha = 0.8, size = 3) + geom_smooth(data = tabg[tabg$colours=="0",], method = lm) + ggtitle("Linear regression of the two proteomic datasets") + scale_colour_manual(values = c("Black", "red", "blue", "forestgreen")) + geom_text(hjust = 1, vjust = 1) 

```

There are `r nprot` unique proteins that are present in the two tables. `r nprotnoexp` proteins are regulated in our dataset and are not in the Barra et al. dataset. In total, `r nprotnoexp0` proteins are quantified in our dataset and absent of Barra et al. table. There are `r nInThe3` proteins quantified in teh 3 datasets.