---
title: "Treg/Tconv: comparison of Duguet's data set with expression and proteomic datasets"
author: "Marie Locard-Paulet"
date: '`r Sys.Date()`'
output: html_document
---

```{r pakages, echo=F}
require(ggplot2)
require(reshape2)
require(knitr)
```

Here I want to compare our dataset with previous works. I will work with the supplementary data of two papers:

*Foxp3 Transcription-Factor-Dependent and -Independent Regulation of the Regulatory T Cell Transcriptional Signature. Jonathan A. Hill, Markus Feuerer, Kaley Tash, Sokol Haxhinasto, Jasmine Perez, Rachel Melamed, Diane Mathis, and Christophe Benoist1,*

DOI 10.1016/j.immuni.2007.09.010

*Transcription Factor 7 Limits Regulatory T Cell Generation in the Thymus. Melanie M. Barra, David M. Richards, Jenny Hansson, Ann-Cathrin Hofer, Michael Delacher, Jan Hettinger, Jeroen Krijgsveld and Markus Feuerer*

J Immunol published online 31 August 2015

The table with our MS results is the table (input table) `OutputTables/TotalAnalysisTconvTreg.txt`.

# Comparison with the expression dataset

Hill et al. 2007.

I use the expression data file `Raw/ExpressionData/GSE7460_series_matrix.txt` downloaded from the [Omnibus expression datasets webpage](http://www.ncbi.nlm.nih.gov/geo/). I have computed a ratio of normalised intensities Treg/Tconv (Markdown file `ExpressionAnalysis.Rmd` in the `ExpressionData` folder). All is saved in the file `ExpData.Rdata` that I load (`nmat` contains the fold changes ratio, `mat3` the raw expression values).

* I perform linear regressions first, then I make a graphical representation.
* In the expression table, I indicate wich genes are not present in our MS data set by adding a column called "PresenceInOurDataset": "Present in our MS dataset" means we quantify the corresponding protein; "not in our MS dataset" means that we don't identify the corresponding protein in our MS dataset.
* I then add a column named "MeanExpression" to our restult table that contains the mean expression observed for the corresponding protein in the expression dataset.

The final table is saved as `OutputTables/2PepOrMore/SignatureFoxp3.txt`.


```{r Expression, echo = F}
tab <- read.delim("OutputTables/2PepOrMore/TotalAnalysisTconvTreg.txt", stringsAsFactors = F)
load("Raw/ExpressionData/ExpData.Rdata")

means <- log2(rowMeans(nmat))
tabexp <- nmat

tab1 <- cbind(tab, "MeanExpression"= means[match(tab$Gene.names, names(means))])
tab10 <- tab1
tab1 <- tab1[!is.na(tab1$MeanExpression),]
ngenes <- length(unique(tab1$Gene.names)) # number of unique genes in our dataset

#colours for plot:
col <- rep("Not", nrow(tab1))
col[abs(tab1$MeanExpression) >= 2 | abs(tab1$FC) >= 2] <- "Above 2"
col[tab1$Gene.names=="Themis"] <- "Themis"
text <- rep(NA, nrow(tab1))
text[abs(tab1$MeanExpression) >= 2 | abs(tab1$FC) >= 2] <- tab1$Gene.names[abs(tab1$MeanExpression) >= 2 | abs(tab1$FC) >= 2]
text[tab1$Gene.names=="Themis"] <- "Themis"
tab1 <- cbind(tab1, "col" = col, "lab" = text)

lrd <- lm(tab1$FC~tab1$MeanExpression)
summary(lrd)
par(mfrow = c(2,2))
plot(lrd)

tabg <- tab1

g <- ggplot(data = tabg, aes(y = FC, x = MeanExpression, label = lab)) + geom_vline(xintercept = 0) + geom_hline(yintercept = 0) + geom_smooth(method = lm, fullrange = TRUE) + geom_point(aes(col = col), alpha = 0.7, size = 3) + ggtitle("Expression of Treg markers Vs\nprotein Treg/Tconv ratio in our MS dataset") + theme_classic() + xlab("Mean expression: log2(Treg/Tconv)") + ylab("Proteomic data: log2(Treg/Tconv)") + scale_colour_manual(values = c("red", "Black", "orange")) + geom_text(col = "red", hjust = 0.5, vjust = -0.8, size = 4)
print(g)


pdf("Figures//2PepOrMore//ExpressionCorrelation.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
g
dev.off()

# Proteomic data regulated and not present in the expression dataset:
reg <- tab10[abs(tab10$FC) >= 1 & tab10$adjpval<=0.05,]
nprotnoexp <- nrow(reg[is.na(reg$MeanExpression),])

tabg <- reg[is.na(reg$MeanExpression),]
tabg <- tabg[order(tabg$FC),]
tabg$Gene.names <- factor(as.character(tabg$Gene.names), levels = as.character(tabg$Gene.names))

g <- ggplot(tabg, aes(y = FC, x = Gene.names)) + geom_hline(yintercept = 0) + geom_bar(stat = "identity", width = 0.7) + theme_bw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 9)) + ggtitle("Genes regulated in our MS dataset\nwith no expression data")
print(g)

# Make a column in the expression table that indicates if the gene is present in our MS dataset or not.
Pres <- rep("not in our MS dataset", nrow(nmat))
Pres[rownames(nmat) %in% as.character(tab$Gene.names)] <- "Present in our MS dataset"
exp <- cbind(nmat, "PresenceInOurDataset" = Pres, "Gene.name" = row.names(nmat))
#write.table(exp, "OutputTables//2PepOrMore/signatureFoxp3.txt", sep = "\t", row.names = F)
npresentInBoth <- length(unique(rownames(exp)[exp[,4]=="Present in our MS dataset"]))
npresentInExpOnly <- length(unique(rownames(exp)[exp[,4]!="Present in our MS dataset"]))
kable(data.frame(table(Pres))) # numbers of genes in the expression dataset that are present or not in our dataset.
```

There are `r ngenes` unique genes that are present in the expression table and quantified in our MS dataset. There are `r nprotnoexp` genes that are regulated in our dataset and not present in the expression dataset. These are presented in the previous figure.

The table `OutputTables/2PepOrMore/signatureFoxp3.txt` contains the initial expression data with an additional column stating if the gene is quantified in our MS dataset.

The table `OutputTables/2PepOrMore/TableAll.txt` contains our result table with the additional columns of mean in each dataset, including a `mapping` column that is the first Majority protein ID that I use to merge the other datasets to ours (see the following section).

# Comparison with the proteomic dataset

Barra et al. 2015.

I use the Supplemental Table 1: Calculated dataset of proteins differentially expressed between Treg cells and Tconv cells.

I use all the protein accessions in the column `Accession` as input in [Uniprot ID mapper](http://www.uniprot.org/uploadlists/) to get all possible protein accessions of these proteins: `Raw/ProteomicsData/uniprot-yourlist%3AM2015113099G02LBX6Z-GenesMSinput.tab`.

```{r Proteomic, echo=F}
exp <- read.delim("Raw/ProteomicsData/Sup Table 1.csv", sep = ";", header = T)

ProtMap <- sapply(as.character(exp$Accession), function(x) strsplit(x, ",", fixed = T)[[1]][1])
# Issues with retrieval of the old Uniprot accessions so I use the gene names instead:
ProtMap <- as.character(exp$Gene.name)
tabmap <- sapply(tab10$Majority.protein.IDs, function(x) strsplit(x, ";")[[1]][1])
tab10 <- cbind(tab10, "mapping" = tabmap)

maptab <- read.delim("Raw/ProteomicsData/uniprot-yourlist%3AM2015113099G02LBX6Z-GenesMSinput.tab", sep = "\t")
maptab2 <- maptab[maptab[,2] %in% tab10$mapping,]

val <- exp$Average.log2.Ratio.Treg.Tconv[match(maptab2[,1], toupper(exp$Gene.name))]
val2 <- val[match(tab10$mapping, maptab2[,2])]

tabProt <- exp[,c(1:12, 35)]

tab1 <- cbind(tab10, "ProteomicRatioTregTconv"= val2)

tab11 <- tab1
tab1 <- tab1[!is.na(tab1$ProteomicRatioTregTconv),]
ngenes <- length(unique(tab1$Gene.names))
nprot <- length(unique(tab1$mapping))

#colours for plot:
col <- rep("Not", nrow(tab1))
col[abs(tab1$ProteomicRatioTregTconv) >= 2 | abs(tab1$FC) >= 2] <- "Above 2"
text <- rep(NA, nrow(tab1))
text[abs(tab1$ProteomicRatioTregTconv) >= 2 | abs(tab1$FC) >= 2] <- tab1$Gene.names[abs(tab1$ProteomicRatioTregTconv) >= 2 | abs(tab1$FC) >= 2]
tab1 <- cbind(tab1, "col" = col, "lab" = text)

lr <- lm(tab1$FC~tab1$ProteomicRatioTregTconv)
summary(lr)
par(mfrow = c(2,2))
plot(lr)

tabg <- tab1

g <- ggplot(data = tabg, aes(y = FC, x = ProteomicRatioTregTconv, label = lab)) + geom_vline(xintercept = 0) + geom_hline(yintercept = 0) + geom_smooth(method = lm, fullrange = TRUE) + geom_point(aes(col = col), alpha = 0.7, size = 3) + ggtitle("Treg/Tconv ratio in the two MS datasets") + theme_classic() + xlab("Proteomic data from Barra el al. 2015: log2(Treg/Tconv)") + ylab("Proteomic data: log2(Treg/Tconv)") + scale_colour_manual(values = c("red", "Black")) + geom_text(col = "red", hjust = 0.5, vjust = -0.8, size = 4)
print(g)

pdf("Figures//2PepOrMore//ProteomicsCorrelation.pdf", useDingbats=FALSE, 11.69, 8.27) # width and height in inches.
g
dev.off()

#write.table(tab11, "OutputTables/2PepOrMore/TableAll.txt", sep = "\t", row.names = F)

# Proteomic data regulated and not present in the Barra et al. dataset:
reg <- tab11[abs(tab10$FC) >= 1 & tab11$adjpval<=0.05,]
nprotnoexp <- nrow(reg[is.na(reg$ProteomicRatioTregTconv),])
nprotnoexp0 <- nrow(tab11[is.na(tab11$ProteomicRatioTregTconv),])

tabg <- reg[is.na(reg$ProteomicRatioTregTconv),]
tabg <- tabg[order(tabg$FC),]
tabg$Gene.names <- factor(as.character(tabg$Gene.names), levels = as.character(tabg$Gene.names))

g <- ggplot(tabg, aes(y = FC, x = Gene.names)) + geom_hline(yintercept = 0) + geom_bar(stat = "identity", width = 0.7) + theme_bw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 10)) + ggtitle("Proteins regulated in our MS dataset\nwith no quantification in Barra et al.")
print(g)

# Make a column in the expression table that indicates if the gene is present in our MS dataset or not.
Pres <- rep("not in our MS dataset", nrow(exp))
Pres[toupper(exp$Gene.name) %in% as.character(maptab2[,1])] <- "Present in our MS dataset"
exp <- cbind(exp, "PresenceInOurDataset" = Pres)
####write.csv(exp, "Papier JI 2015//Sup Table 1-V2.csv", row.names = F)
npresentInBoth <- length(unique(exp$Gene.name[exp$PresenceInOurDataset=="Present in our MS dataset"]))
npresentInExpOnly <- length(unique(exp$Gene.name[exp$PresenceInOurDataset!="Present in our MS dataset"]))

kable(data.frame(table(Pres)))

nInThe3 <- nrow(tab11[!is.na(tab11$MeanExpression) & !is.na(tab11$ProteomicRatioTregTconv),])
```

There are `r npresentInBoth` unique genes that are present in the two tables (`r npresentInExpOnly` are only identified in Barra et al. dataset). `r nprotnoexp` genes are regulated in our dataset and are not in the Barra et al. dataset. In total, `r nprotnoexp0` genes are quantified in our dataset and absent of Barra et al. table. There are `r nInThe3` genes quantified in the 3 datasets.

# Creation heatmap with the three datasets.

In our dataset, there are `r length( tab$Gene.names[tab$Gene.names==""])` rows with no gene names, 3 corresponding to H2-D1 and 2 for Tmpo (these are different isoforms, in the case of Tmpo they don't present the same abundancy).

For this figure, I manually remove the row with no gene name, and corresponding to `Majority.protein.IDs` == "P14427" and "P01900;P01895" for H2-D1 (highest pval, no adjusted pval); "Q61029" for Tmpo (highest pval, no adjusted pval).

In the proteomic data table, some isoforms remain (duplicated gene names). So I use `Accession` as row names. For each gene, I keep the quan value corresponding to the SwissProt identification with the maximum number of Unique + razor peptides.

For the tilemap, I select the genes that have a mean FC >= 1 or <= -1 in 2 datasets. Then, I present 50% of the top hits ordered by mean FC: `Figures/TregTconv3DS.pdf`.

```{r, echo=F}
table(tab$Gene.names)[table(tab$Gene.names)>1]
tab <- tab[tab$Majority.protein.IDs != "P14427" & tab$Majority.protein.IDs != "P01900;P01895" & tab$Majority.protein.IDs != "Q61029",]
tab <- tab[tab$Gene.names!="",]
mat <- matrix(ncol = 7, nrow = nrow(tab))
for(i in 1:7) {
  mat[,i] <- as.numeric(tab[,(11+i)]) - as.numeric(tab[,(4+i)])
}

rownames(mat) <- tab$Gene.names
colnames(mat) <- c("M11_R1", "M11_R2", "M12_1", "M12_2", "M12_3", "M5", "M6")
tab4 <- data.frame(mat, "Gene.names"=rownames(mat))
tabexp1 <- as.data.frame(log2(tabexp))
tabexp1 <- cbind(tabexp1, "Gene.names" = rownames(tabexp))
tab2 <- merge(tab4, tabexp1, by="Gene.names", all = T)
#tabProt <- tabProt[!is.na(tabProt$Stdev.log2.Ratio.Treg.Tconv),]
tabProt <- tabProt[tabProt$Gene.name!="",]
# Remove gene duplicates:
t <- table(tabProt$Gene.name)
dup <- names(table(tabProt$Gene.name))[table(tabProt$Gene.name)>1]
keep <- vector(length = nrow(tabProt))
keep[!(tabProt$Gene.name %in% dup)] <- TRUE
sub <- tabProt[tabProt$Gene.name %in% dup,]
g <- sapply(as.character(unique(sub$Gene.name)), function(x) sub$Razor...unique.Peptides[sub$Gene.name==x])
g <- sapply(g, max)
sub2 <- tabProt[keep==TRUE,]
for(i in 1:length(g)) {
  sub2 <- rbind(sub2, as.vector(sub[(sub$Gene.name==names(g)[i] & sub$Razor...unique.Peptides==g[i]),]))
}
tabProt <- sub2

names(tabProt)[3] <- "Gene.names"
tab2 <- merge(tab2, tabProt, by="Gene.names", all = T)

mat <- tab2[,c(2:11,18:21)]

par(mar = c(13, 4, 5, 1))
boxplot(mat, las = 2, main = "Treg/Tconv in the three datasets")
require(gplots)

maxval <- max(mat, na.rm=T)
minval <- min(mat, na.rm=T)
collim <- max(abs(maxval), abs(minval))

df2 <- data.frame(mat, "Gene.names"=tab2$Gene.names)
meanUs <- rowMeans(mat[,1:7], na.rm = T)
meanProt <- rowMeans(mat[,11:14], na.rm = T)
meanExp <- rowMeans(mat[,8:10], na.rm = T)
mat2 <- cbind(meanUs, meanProt, meanExp)
rownames(mat2) <- rownames(mat)
par(mar = c(7, 4, 5, 1))
boxplot(mat2, las = 2, main = "Mean for each protein in\nthe 3 data sets")

NAUs <- abs(meanUs)>=1; NAProt <- abs(meanProt)>=1; NAExp <- abs(meanExp)>=1
NAUs[is.na(NAUs)] <- FALSE; NAProt[is.na(NAProt)] <- FALSE; NAExp[is.na(NAExp)] <- FALSE
mat3 <- df2[sapply(1:length(NAUs), function(x) sum(NAUs[x], NAProt[x], NAExp[x], na.rm = T)) >= 2,] # Select the genes regulated in a minimum of 2 datasets
mat3 <- mat3[!is.na(mat3$Gene.names),]
df3 <- data.frame(mat2, "Gene.names"=df2$Gene.names)

# select the 50% up and down:
s <- rowMeans(mat3[,1:14], na.rm = T)
down <- length(s[s<0]) %/% 2 
up <- length(s[s>0]) %/% 2
mat3 <- mat3[order(s),]
mat3 <- mat3[c(1:down, (nrow(mat3)-up+1):nrow(mat3)),]

colours = c("darkblue", "yellow", "orange")
colours = c("green4", "yellow", "red")
#colours = c("darkblue", "green", "yellow", "red")
#colscale = seq(-collim, collim, length = 5)
#colscale = seq(-8.3, 9.5, length = 4)
colscale = c(-9.5, -0.5, 0.5, 9.5)



require(reshape2)
df <- melt(data.frame(mat3))
df <- df[!is.na(df$variable),]
df <- df[!is.na(df$Gene.names),]
df$Gene.names <- factor(as.character(mat3$Gene.names), levels = as.character(mat3$Gene.names)[order(rowMeans(mat3[,1:14], na.rm = T))])
g <- ggplot(df, aes(y = Gene.names, x = variable)) + theme_classic()+ geom_tile(aes(fill = value)) + scale_fill_gradientn(colours = colours, values = colscale, rescaler = function(x, ...) x, oob = identity) + theme(axis.title.x = element_blank()) + theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), axis.text.y = element_text(size = 10)) + geom_vline(xintercept = c(7.5, 10.5), size = 1) + ggtitle("Our proteomic dataset, transcriptomics and proteomics\nordered by mean FC")#+ geom_text(aes(label=round(value, digits = 3)), size = 3)
g

pdf("Figures/2PepOrMore/TregTconv3DS.pdf", useDingbats=FALSE, 8.27, 11.69) # width and height in inches.
g
dev.off()

#write.table(mat3, "OutputTables/2PepOrMore/HeatMapProtExp_Table.txt", sep = "\t", row.names = F)
```

The table used for the final heatmap is: `OutputTables/2PepOrMore/HeatMapProtExp_Table.txt`.

```{r}
sessionInfo()
```

